pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -184,6 +184,7 @@ public enum AggregationFunctionType {
       OperandTypes.family(List.of(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER, SqlTypeFamily.BOOLEAN), i -> i == 2),
       SqlTypeName.OTHER),
   LISTAGG("listAgg", SqlTypeName.OTHER, SqlTypeName.VARCHAR),
```

Comment:
We'll need to update this to use a custom operand type checker and return type inference